crates/bevy_text/src/text.rs

```diff
@@ -174,20 +174,69 @@ pub struct TextStyle {
     ///
     /// A new font atlas is generated for every combination of font handle and scaled font size
     /// which can have a strong performance impact.
-    pub font_size: f32,
+    pub font_size: FontSize,
     pub color: Color,
 }
 
 impl Default for TextStyle {
     fn default() -> Self {
         Self {
             font: DEFAULT_FONT_HANDLE.typed(),
-            font_size: 12.0,
+            font_size: FontSize::Default,
             color: Color::WHITE,
         }
     }
 }
 
+/// The default font size.
+/// Used when the size field of TextStyle is set to Default.
+#[derive(Resource, Clone, Copy, Debug)]
+pub enum DefaultFontSize {
+    /// Default font size in logical pixels
+    Px(f32),
```

Comment:
Should there be any notes here about reasonable values? e.g., negative values?

Reply:
I think it's implicit that negative values would be nonsense but we could add some sort of error message. In main atm if `font_size` is set to a negative we just fail silently and draw nothing. Perhaps there should be a warning there too.